### ISSUE TITLE
Mass message circulation between nodes

### ISSUE THREAD

**Issue by MikaelElkiaer** [nats-io/nats-server#4416]
### What version were you using?

v2.9.15 initially, then upgraded to v2.9.21

### What environment was the server running in?

- Running in an AWS cluster with at least 7 nodes - each with 8 cores and 30 GB RAM.
- Installed via the official chart - initially 0.19.12 along with server v2.9.15, then upgraded to 1.0.0 along with server v2.9.21.
- 3 replicas, but scaled down to 1 to mitigate the problem, then scaled back up again to 3 in order to investigate.
- No JetStream enabled
- Websocket and leaf node enabled
- Using full resolver with pre-seeded keys

### Is this defect reproducible?

Not easily so - we cannot force it, but usually see it start around a server or service upgrade/restart.

More details in this Slack thread: https://natsio.slack.com/archives/C069GSYFP/p1691403718768509

Last message in the thread included here for visibility:

> I've messed around a bit now, and have already made some interesting observations.
Please see the attached screenshot.
Up and till 1) I was running a single replica nats server v2.9.15.
From 1) that was upscaled to 3 replicas [and version v2.9.21] - interestingly the messages per second went up after being consistent at 0-1.
At 2) I did a rolling restart of our queue group loggers and afterwards a rolling restart of our nats servers.
Not a big change, but noticeable.
Between 2) and 3) I discovered that our queue group loggers were subscribing to a subject that was no longer in use - as the subject had been prefixed, causing the queue group loggers to subscribe to an unused subject.
I updated the queue group loggers at 3) to now subscribe to the updated subject, causing them once again to recieve messages.
What is really interesting to me, is how much traffic there is between nats server pods, for the unsubscribed topic.

![image](https://github.com/nats-io/nats-server/assets/2102306/6eefa8c1-842f-499c-ae37-ae33903f934a)

### Given the capability you are leveraging, describe your expectation?

I expected the same, or perhaps a lowered amount of messages.

### Given the expectation, what is the defect you are observing?

One or few messages are sent indefinitely between nodes - only limited by CPU or network.

**Comment by derekcollison** [comment]
What is the overall topology? 

Do you use leafnode connections? How many and what topology?
Do you do any account exports/imports?
Do you use DQs? I think yes from above. Where are they running?

**Comment by MikaelElkiaer** [comment]
Damn, had hoped you would look at my Slack thread.
I will add some details here for simplicity.

1. We use leafnodes for any connected external machine that is running a .NET application.
Could be tens of machines.
2. We do lots of exports/imports. We have an account for backend services to communicate, and an account for each tenant. Certain tenant subjects are mapped to the backend account for cross-tenant matters. We have less than 10 tenants and less than 10 backend services.
3. DQs? Is that queue groups? If so, yes, we have a couple of jobs that pick up messages on certain topics, outputting them to stdout so they can be picked up by Grafana Loki.

![image](https://github.com/nats-io/nats-server/assets/2102306/9e2db40f-5bd6-4f40-ac1e-fdf602ddd051)

**Comment by MikaelElkiaer** [comment]
It might be useful, so here is another metric simply counting nats log messages (with debug and trace enabled) for the same time window as the graph above.

![MicrosoftTeams-image (2)](https://github.com/nats-io/nats-server/assets/2102306/d7600c43-f8fc-43f2-bd2f-9b8bb96ee203)

**Comment by derekcollison** [comment]
Did 2.10.2 fix?

**Comment by MikaelElkiaer** [comment]
We did not upgrade yet.
Also, we haven't seen the issue since I reported this.
It seems that something we did has fixed the issue for us, or at least decreased the chances of it happening.

Basically, the 3 actions from the issue description were appiled to our DEV environment.
It ran there for a month or two, without the issue re-appearing.
We have since then applied it to STG and PROD, more than a month ago, and have not seen the issue there either.
